mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -512,6 +513,92 @@ object Vectors {
     squaredDistance
   }
 
+  private def dot(a : DenseVector, b : DenseVector) : Double = {
+    (a.toArray zip b.toArray).map(x => (x._1 * x._2)).sum
+  }
+
+  private def dot(a : SparseVector, b : DenseVector) : Double = {
+    (a.indices zip a.values).map(x => { b(x._1)*x._2 }).sum
```

Comment:
problem is that toBreeze is private ( i don't think it should be but i felt i was overstepping to change that ). Also its just a bit annoying from programmers perspective. I can see what you are saying, I just think that extra step should be inside rather than outside. Really I think all the breeze operations should be defined for these vectors (or just use breeze vectors but I guess a single vector may need to be distributed).